cpp/src/parquet/arrow/reader.cc

```diff
@@ -485,8 +485,9 @@ class LeafReader : public ColumnReaderImpl {
         NextRowGroup();
       }
     }
-    RETURN_NOT_OK(
-        TransferColumnData(record_reader_.get(), field_, descr_, ctx_->pool, &out_));
+    RETURN_NOT_OK(TransferColumnData(record_reader_.get(),
+                                     input_->column_chunk_metadata(), field_, descr_,
```

Comment:
The call to `input_->column_chunk_metadata()` fails here if the list of `row_groups` in `input_` is empty, because `input_` is not yet initialized properly at this point in that case

Via `row_group_metadata()` -> `RowGroup(-1)`

Reply:
Good catch!
Could you open a new issue for it?

Reply:
#45339